contacts.module

```diff
@@ -160,6 +161,30 @@ function template_preprocess_contacts_dash_summary(array &$variables) {
   }
 }
 
+/**
+ * Pre render callback for dashboard blocks to add edit link to title.
+ *
+ * @param $element
+ *   The block render array.
+ *
+ * @return array
+ *   The block render array.
+ */
+function contacts_dashboard_block_edit_link_pre_render($element) {
+  if (!empty($link = $element['#dashboard_label_edit_link'])) {
+    if ($element['content']['#title']) {
+      $element['content']['#title'] = new FormattableMarkup('@label [@link]', [
+        '@label' => $element['content']['#title'],
+        '@link' => $link->toString(),
+      ]);
+    }
+    else {
+      $element['content']['#title'] = $link->toString();
```

Comment:
Hmm... What does it do if we have no title or aren't displaying it? Have we thought about the formatting etc? Also raises the question of whether the title is actually the appropriate place for this, or if we're better thinking of a specific UI?

Reply:
Ok, there are several config options to configure:

label_display: ('visible' or '0') - if set to 0, not title will be shown and no edit link shown
edit_link: ('title' or 'content') - if we are not showing a title, this should be set to content.
label: if this is empty but label display is set to visible then we get a default title from $block->build().

So I think we actually have all bases covered. We would have to explicitly set the '#title' attribute to NULL for the link to be rendered on its own, which is either on purpose (to only show the edit link) or hiding it wrong.

Reply:
Ok - I'm happy for now. Let's make sure we have a general UI review of the edit links at some point.